be/Middleware/Validation/ValidationAbstract.php

```diff
@@ -0,0 +1,23 @@
+<?php
```

Comment:
Tutaj zrobiłem taką klasę abstrakcyjną która ma już __invoke, a te które dziedziczą z niej muszą tylko zimplementować funkcje validate. Chciałem zrobić to jakoś reużywalne, żeby nie trzebabyło wszędzie tego opakowywać.
Co sądzisz? Czy może to jakąś fabryką by się dało czy jakoś inaczej ?

Reply:
Moim zdaniem zrobiłeś to dobrze :). Choć jedna uwaga a propos architektury klas `Validator`. Moim zdaniem są zbyt mocno zależne od kontrolerów. Tak naprawdę nie ma możliwości skorzystania z nich w innym miejscu.
Zamiast tego, można te klasy przerobić aby były reużywalne. Na przykład `ValidationAddUser` można podzielić na:
- `ValidationUserCodeUnique` (Jako param w jakiś sposób otrzymywałby wartość i sprawdzał, czy kod usera jest unikalny.)
- `ValidationStringLength` (Jako param wartość, jako drugi param długość. No i sprawdzanie :).)

I wtedy można zrobić tak:
```php
$app->post('/users', [UsersCtrl::class, 'addUser'])
    ->add(ValidationStringLength::class) // albo `->add(ValidationUserCodeLength::class)`
    ->add(ValidationUserCodeUnique::class)
    ->add(PrivilegesIsLogged::class);
```

ALBO, co może być jeszcze lepsze, w `add()` dawać jakąś funkcję, która pobiera z Requesta odpowiednią wartość, i przesyła do instancji obiektu `Validation`.

Powyższe można ewentualnie jako osobny PR, gdzie jeden PR będzie na jedną klasę `Validator` :).

Reply:
Spoko, tak też przechodziło mi przez myśl jak to robiłem że można zrobić takie małe walidatory a potem tylko sobie komponować z nich. Ale jednak podszedłem do tego że poprostu każdy serwis ma specjalnie dedykowany swój. Ale myśle że to spoko pomysł, można będzie wdrożyć jak będą chęci :).